lib/store.go

```diff
@@ -0,0 +1,39 @@
+package lib
+
+import "github.com/ethereum/go-ethereum/common"
+
+// Store stores payloads and retrieves them based on blockHash hashes
+type Store interface {
+	Get(blockHash common.Hash) *ExecutionPayloadWithTxRootV1
+	Set(blockHash common.Hash, payload *ExecutionPayloadWithTxRootV1)
+}
+
+// map[common.Hash]*ExecutionPayloadWithTxRootV1
+// map blockHash to ExecutionPayloadWithTxRootV1. TODO: this has issues, in that blockHash could actually be the same between different payloads
+// TODO: clean this up periodically
+
+type store struct {
+	payloads map[common.Hash]*ExecutionPayloadWithTxRootV1
+}
+
+// NewStore creates an in-mem store
+func NewStore() Store {
+	return &store{map[common.Hash]*ExecutionPayloadWithTxRootV1{}}
+}
+
+func (s *store) Get(blockHash common.Hash) *ExecutionPayloadWithTxRootV1 {
+	payload, ok := s.payloads[blockHash]
```

Comment:
Do we need to have locks around these for concurrency?

Reply:
Good catch, thanks! Fixed in https://github.com/flashbots/mev-boost/commit/c7be3e608b3b3979e724620455253a9fbf969c1b